statistics/statistics.go

```diff
@@ -280,56 +280,129 @@ func (p *Plugin) GetConfigPolicy() (*cpolicy.ConfigPolicy, error) {
 	cp.Add([]string{""}, config)
 
 	return cp, nil
+
+}
+
+type byTimestamp []plugin.MetricType
+
+func (sa byTimestamp) Len() int {
+	return len(sa)
+}
+
+func (sa byTimestamp) Less(i, j int) bool {
+	return sa[i].Timestamp().Before(sa[j].Timestamp())
+
+}
+func (sa byTimestamp) Swap(i, j int) {
+	sa[i], sa[j] = sa[j], sa[i]
+}
+
+func (p *Plugin) insertInToTimeBuffer(metric plugin.MetricType, times []time.Time) []time.Time {
+	if p.bufferCurSize == 0 {
+		times[0] = metric.Timestamp()
+	} else {
+		times[p.bufferIndex] = metric.Timestamp()
+	}
+	return times
+}
+
+func (p *Plugin) getTimes(times []time.Time) (time.Time, time.Time) {
+	//log.Printf("Buffer index: %v", metric.Timestamp())
+	if p.bufferCurSize == p.bufferMaxSize && p.bufferIndex != p.bufferMaxSize-1 {
```

Comment:
We need to make sure that we are tracking the second to last index so that when we return p.bufferIndex+1 we are not causing an out of index error.